client/src/components/RelatedItems/RelatedProducts/RelatedProducts.jsx

```diff
@@ -0,0 +1,248 @@
+    setIsModalOpen(false);
+  };
+  const handleCancel = () => {
+    setIsModalOpen(false);
+  };
+
+  const renderFeature = () => {
+    let str = '';
+    for (let key in features) {
+      str += `<div
+            style="
+              display: flex;
+              align-items: center;
+              justify-content: space-around
+            "
+          >
+            <span style="width:33%;text-align:center">${features[key][0]}</span>
+            <h4 style="width:33%;text-align:center">${key}</h4>
+            <span style="width:33%;text-align:center">${features[key][1]}</span>
+          </div>`;
+    }
+    return { __html: str };
+  };
+
+  return (
+    <>
+      <h4>RELATED PRODUCTS</h4>
+      <Swiper
+        modules={[Virtual, Navigation]}
+        onSwiper={setSwiperRef}
+        slidesPerView={3}
+        spaceBetween={50}
+        pagination={{
+          type: 'fraction',
+        }}
+        navigation={true}
+        virtual
+      >
+        {slides.map((slideContent, index) => (
+          <SwiperSlide key={slideContent} virtualIndex={index}>
+            <div
+              style={{
+                height: '380px',
+                width: '250px',
+
+                border: 'solid',
+                display: 'flex',
+                flexDirection: 'column',
+              }}
+            >
+              <div
+                style={{
+                  height: '250px',
+                  backgroundColor: 'grey',
+                  position: 'relative',
+                  overflow: 'hidden',
+                  backgroundImage: `url(${slideContent.photos[0].url})`,
+                  backgroundSize: 'cover',
+                  backgroundRepeat: 'no-repeat',
+                }}
+              >
+                <StarOutlined
+                  style={{
+                    position: 'absolute',
+                    top: '10px',
+                    right: '10px',
+                    cursor: 'pointer',
+                    color: 'white',
+                  }}
+                  onClick={() => handleCompare(slideContent.id)}
+                />
+                {/* <img src={slideContent.photos[0].url} alt="" /> */}
+              </div>
+              <div
+                style={{
+                  flex: 1,
+                  display: 'flex',
+                  flexDirection: 'column',
+                  justifyContent: 'space-evenly',
+                  alignItems: 'flex-start',
+                  padding: '0 10px',
+                }}
+              >
+                <span style={{ fontWeight: 400, fontSize: '15px' }}>
+                  categroy
+                </span>
+                <span style={{ fontWeight: 800 }}>
+                  {slideContent.name.trim()}
+                </span>
+                <span style={{ fontSize: '12px' }}>
+                  ${slideContent.original_price}
+                </span>
+                <Rate allowHalf defaultValue={slideContent.averageRate} />
```

Comment:
Is this a star rating component provided by antd? Please use the one provided as a shared component in the app so the style can be consistent. Also it doesn't looks like the antd rating component allows for things like 3/4 stars which are in business requirements.

For reference the shared star rating component can be found in `client/src/components/common/ReviewStars.js`